cognite/client/_api/datapoints.py

```diff
@@ -959,27 +965,189 @@ def retrieve_dataframe(
+        grans_given = {q.granularity for q in fetcher.all_queries}
+        is_limited = any(q.limit is not None for q in fetcher.all_queries)
+        if fetcher.raw_queries or len(grans_given) > 1 or is_limited:
+            raise ValueError(
+                "Cannot return a uniform index when asking for aggregates with multiple granularities "
+                f"({grans_given}) OR when (partly) querying raw datapoints OR when a finite limit is used."
+            )
+
         df = fetcher.fetch_all_datapoints_numpy().to_pandas(
             column_names, include_aggregate_name, include_granularity_name
         )
-        if not uniform_index:
-            return df
-
         start = pd.Timestamp(min(q.start for q in fetcher.agg_queries), unit="ms")
         end = pd.Timestamp(max(q.end for q in fetcher.agg_queries), unit="ms")
         (granularity,) = grans_given
         # Pandas understand "Cognite granularities" except `m` (minutes) which we must translate:
         freq = cast(str, granularity).replace("m", "T")
         return df.reindex(pd.date_range(start=start, end=end, freq=freq, inclusive="left"))
 
+    Aggregates = Literal[
+        "average",
+        "sum",
+        "count",
+        "sum",
+        "interpolation",
+        "stepInterpolation",
+        "continuousVariance",
+        "discreteVariance",
+        "totalVariation",
+    ]
+
+    def retrieve_dataframe_in_tz(
+        self,
+        *,
+        id: int | Sequence[int] | None = None,
+        external_id: str | Sequence[str] | None = None,
+        start: datetime,
+        end: datetime,
+        aggregates: list[Aggregates] | Aggregates | None = None,
+        granularity: Optional[str] = None,
+        ignore_unknown_ids: bool = False,
+        uniform_index: bool = False,
+        include_aggregate_name: bool = True,
+        include_granularity_name: bool = False,
+        column_names: Literal["id", "external_id"] = "external_id",
+    ) -> pd.DataFrame:
+        """Get datapoints directly in a pandas dataframe in the time zone given in start and end.
+
+        **Note** This is a convenicence method. It builds on top of the retrieve_array and retrieve_dataframe methods.
+        It enables you to do correct aggregates in a local time zone with daily, weekly, quartely, and yearly
+        aggregates, including accounting for DST (Daylight Saving Time).
+
+        In short, this method function as follows:
+            1. Get the time zone from start and end.
+            2. Split the time range from start to end into intervals with fixed UTC offset.
+            3. Create a query for each interval and all the retrieve_array method.
+            4. Stack the resulting arrays into a single column in the resulting DataFrame.
+
+        The queries to retrieve_arrays are translated to a multiple of hours. This means that time zones which
+        are not a whole hour offset from UTC are not supported.
```

Comment:
I suggest we make this into a red warning 😄 
```suggestion
        Warning:
            The queries to retrieve_arrays are translated to a multiple of hours. This means that time zones that
            are not a whole hour offset from UTC are not supported (yet). The same is true for time zones that observe DST
            with an offset from standard time that is not a multiple of 1 hour.
```